vhost-user-backend/src/backend.rs

```diff
@@ -89,9 +89,16 @@ pub trait VhostUserBackend: Send + Sync {
     /// Set handler for communicating with the frontend by the gpu specific backend communication
     /// channel.
     ///
-    /// This method only exits when the crate feature gpu-socket is enabled, because this is only
-    /// useful for a gpu device.
-    fn set_gpu_socket(&self, _gpu_backend: GpuBackend);
+    /// This function returns a `Result`, returning an error if the backend does not implement this
+    /// function.
+    /// A default implementation is provided as we cannot expect all backends to implement this
+    /// function.
```

Comment:
nit: I'd remove these 2 lines, I don't see much more value after the lines above,